uwblueprint/sistering
# Polish admin posting pages

## Ticket link

Closes fullcalendar/fullcalendar-react#138



## Implementation description
In addition to completing requirements outlined in the ticket, this PR also...
* Adds new DateTimeUtils functions
* Fixes datetime bugs
* Fixes UI bugs
* Sets current week in shift calendar
* Executes `createShifts` mutation in create posting review page

Please see commit messages for more details.



## Steps to test
1. Run the application locally (avoiding writes to staging DB for now to keep it clean)
```bash
docker-compose up --build
```
2. Go to the create posting page and create a posting
[![Create Posting Demo](https://cdn.loom.com/sessions/thumbnails/6f00d1b414264c40a9ff773dd1b40050-with-play.gif)](https://www.loom.com/share/6f00d1b414264c40a9ff773dd1b40050 "Create Posting Demo")
3. Verify data in the database
```bash
docker exec -it sistering_db/bin/bash -c "psql -U postgres -d sistering"
```
```sql
SELECT * FROM postings;
SELECT * FROM shifts WHERE posting_id = 'INSERT_POSTING_ID';
```



## What should reviewers focus on?
* Please review **commit by commit**
* End-to-end functionality and correctness of behaviour
* Code cleanliness
* General adherence to hi-fi designs (note: another round of refinement is needed to be pixel-perfect, will require a design review)


## Checklist
